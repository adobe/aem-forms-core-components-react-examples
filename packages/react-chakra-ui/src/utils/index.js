import React from 'react';
import { FormContext } from '@aemforms/af-react-renderer';
import { createFormInstance } from '@aemforms/af-core';
import { ChakraProvider } from '@chakra-ui/react'
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';

export const DEFAULT_ERROR_MESSAGE = 'There is an error in the field';

export const createForm = (field) => {
  const formJson = {
    items: [field]
  };
  return createFormInstance(formJson);
};

export const Provider = (form, mappings, locale, dictionaries) => props => {
  const c = {
    form,
    mappings,
    modelId: '$form',
    refMap: {}
  };
  const { children } = props;
  return (
    <ChakraProvider>
      <IntlProvider locale={locale} messages={dictionaries}>
        <FormContext.Provider value={c}>{children}</FormContext.Provider>
      </IntlProvider>
    </ChakraProvider>
  );
};

export const renderComponent = function (Component) {
  const test = (field, operation) => {
    const form = createForm(field);
    if (operation) {
      operation(form, form.items[0]);
    }
    const e = form.items[0].getState();
    let component = <Component {...e} />;
    const wrapper = Provider(form);
    const renderResponse = render(component, { wrapper });
    return {
      renderResponse,
      form,
      element: form?.items[0],
      component
    };
  };
  return test;
};
