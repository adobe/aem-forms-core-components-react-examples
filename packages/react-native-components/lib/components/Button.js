"use strict";
/*************************************************************************
* ADOBE CONFIDENTIAL
* ___________________
*
* Copyright 2023 Adobe
* All Rights Reserved.
*
* NOTICE: All information contained herein is, and remains
* the property of Adobe and its suppliers, if any. The intellectual
* and technical concepts contained herein are proprietary to Adobe
* and its suppliers and are protected by all applicable intellectual
* property laws, including trade secret and copyright laws.
* Dissemination of this information or reproduction of this material
* is strictly forbidden unless prior written permission is obtained
* from Adobe.

* Adobe permits you to use and modify this file solely in accordance with
* the terms of the Adobe license agreement accompanying it.
*************************************************************************/
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const native_base_1 = require("native-base");
const react_1 = __importDefault(require("react"));
const withRuleEngine_1 = __importDefault(require("../shared/withRuleEngine"));
const ButtonComponent = function (props) {
    const { label, enabled, dispatchClick } = props;
    return react_1.default.createElement(native_base_1.Button, { onPress: dispatchClick, isDisabled: enabled === false }, label === null || label === void 0 ? void 0 : label.value);
};
exports.default = (0, withRuleEngine_1.default)(ButtonComponent);
