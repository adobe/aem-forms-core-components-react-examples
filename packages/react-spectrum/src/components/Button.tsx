/*
 * Copyright 2022 Adobe, Inc.
 *
 * Your access and use of this software is governed by the Adobe Customer Feedback Program Terms and Conditions or other Beta License Agreement signed by your employer and Adobe, Inc.. This software is NOT open source and may not be used without one of the foregoing licenses. Even with a foregoing license, your access and use of this file is limited to the earlier of (a) 180 days, (b) general availability of the product(s) which utilize this software (i.e. AEM Forms), (c) January 1, 2023, (d) Adobe providing notice to you that you may no longer use the software or that your beta trial has otherwise ended.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL ADOBE NOR ITS THIRD PARTY PROVIDERS AND PARTNERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { Button } from '@adobe/react-spectrum';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import React from 'react';
import {FieldJson, State} from '@aemforms/af-core';
import {
    baseConvertor,
    combineConvertors,
    richTextString
} from '../utils/SpectrumMappers';
import {useRenderer, WithViewState} from '@aemforms/af-react-renderer';

const mapper = combineConvertors((a: WithViewState<FieldJson>, b, f) => {
        const localizedLabel = f('label.value');
        return {
            variant :'primary',
            children : a.label?.visible === false || !a.label?.value ? '' : (a.label?.richText === true ? richTextString(localizedLabel) : localizedLabel),
            onPress : b.dispatchClick
        };
    }, baseConvertor);


const ButtonFormComponent = (field: State<FieldJson>) => useRenderer(field, Button, mapper);

export default ButtonFormComponent;