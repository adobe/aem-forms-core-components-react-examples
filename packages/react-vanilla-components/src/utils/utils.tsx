export const formatBytes = (bytes: number, decimals = 0) => {
    if (!+bytes) {
      return '0 Bytes';
    }
    const k = 1024;
    const dm = decimals < 0 ? 0 : decimals;
    const sizes = ['bytes', 'kb', 'mb', 'gb', 'tb', 'pb'];
    const i = Math.floor(Math.log(bytes) / Math.log(k));

    return `${parseFloat((bytes / Math.pow(k, i)).toFixed(dm))} ${sizes[i]}`;
  };

  export const syncAriaDescribedBy = (id: string, tooltip?: string, description?: string, error?: string) => {
    const descriptions = [];

    if (description) {
        descriptions.push(`${id}__longdescription`);
    }
    if (tooltip) {
        descriptions.push(`${id}__shortdescription`);
    }
    if (error) {
        descriptions.push(`${id}__errormessage`);
    }

    return descriptions.join(' ');
};