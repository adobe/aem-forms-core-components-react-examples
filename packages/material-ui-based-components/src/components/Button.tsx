import React, { useCallback } from 'react';
import { withRuleEngine } from '../shared/withRuleEngine';
import { PROPS } from '../utils/types';
import Button from '@mui/material/Button';

const ButtonComponent = (props: PROPS) => {

  const { label, enabled, id, name } = props;

  const clickHandler = useCallback((event: any) => {
    props.dispatchClick(event);
  }, [props.dispatchClick]);

  return (
    <Button variant={props.layout?.variant} disabled={!enabled} onClick={clickHandler} id={id} name={name} color={props.layout?.color}>
      {label?.visible ? label.value : ''}
    </Button>
  );
};
export default withRuleEngine(ButtonComponent);