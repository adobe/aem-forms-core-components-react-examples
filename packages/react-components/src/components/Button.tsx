import React from 'react';
import {State, FieldJson} from '@aemforms/af-core';
import withRuleEngine from '../utils/HOC';
import { IHandler } from '../utils/type';

const Button = (props: State<FieldJson & IHandler>) => {
   const { label, enabled } = props;
   return (
    <div className="cmp-adaptiveform-button" >
       <button className="cmp-adaptiveform-button__widget" aria-label="Button" disabled={!enabled} onClick={props.dispatchClick()}>
          {label?.visible && <span className="cmp-adaptiveform-button__text">{label.value}</span>}
       </button>
    </div>
   );
};

export default withRuleEngine(Button);

