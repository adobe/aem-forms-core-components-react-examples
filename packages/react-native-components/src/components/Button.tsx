/*************************************************************************
* ADOBE CONFIDENTIAL
* ___________________
*
* Copyright 2023 Adobe
* All Rights Reserved.
*
* NOTICE: All information contained herein is, and remains
* the property of Adobe and its suppliers, if any. The intellectual
* and technical concepts contained herein are proprietary to Adobe
* and its suppliers and are protected by all applicable intellectual
* property laws, including trade secret and copyright laws.
* Dissemination of this information or reproduction of this material
* is strictly forbidden unless prior written permission is obtained
* from Adobe.

* Adobe permits you to use and modify this file solely in accordance with
* the terms of the Adobe license agreement accompanying it.
*************************************************************************/

import { Button } from 'native-base';
import React from 'react';
import { PROPS } from '../utils/types';
import withRuleEngine from '../shared/withRuleEngine';

const ButtonComponent = function (props: PROPS) {
  const { label, enabled, dispatchClick } = props;
  return <Button onPress={dispatchClick} isDisabled={enabled === false} {...props.layout}>{label?.value}</Button>;
};
export default withRuleEngine(ButtonComponent);