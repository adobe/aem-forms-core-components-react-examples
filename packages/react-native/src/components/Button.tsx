import { Button } from 'native-base';
import { State, FieldJson } from '@aemforms/af-core';
import { useRenderer } from '@aemforms/af-react-renderer';
import { baseConvertor, combineConvertors, buttonConvertor } from '../utils/mappers';

const ButtonComponent = function (props: State<FieldJson>) {
  return props.visible ? useRenderer(props, Button, combineConvertors(baseConvertor, buttonConvertor)) : null;
};
export default ButtonComponent;