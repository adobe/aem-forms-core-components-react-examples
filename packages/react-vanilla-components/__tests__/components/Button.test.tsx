import Button from '../../src/components/Button';
import { renderComponent } from '../utils';

const field = {
  name: 'button',
  label: {
    value: 'button',
  },
  fieldType: 'button',
  visible: true
};
const helper = renderComponent(Button);

describe('Button', () => {

  test('should render with label', () => {
    const f = {
      ...field,
    };
    const { renderResponse } = helper(f);
    const button = renderResponse.getByText(field.label.value);
    expect(button).toBeTruthy();
  });

  test('it should handle visible property', () => {
    const f = {
      ...field,
      visible: false,
    };
    const { renderResponse } = helper(f);
    const button = renderResponse.queryByText(field.label.value);
    expect(button).toBeNull();
  });

  test('html in the label should be handled for non rich text', async () => {
    const f = {
      ...field,
      label: {
        value: '<p>title inside p tags</p>',
        richText: true,
        visible: true
      }
    }
    let { renderResponse } = await helper(f);
    expect(renderResponse.container.innerHTML).toContain('<p>title inside p tags</p>');
  });

});