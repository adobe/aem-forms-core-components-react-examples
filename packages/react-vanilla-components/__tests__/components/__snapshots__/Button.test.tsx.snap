// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `
<div>
  <div
    class="cmp-adaptiveform-button "
  >
    <div
      class="cmp-adaptiveform-checkbox__help-container"
    />
    <button
      aria-label=""
      class="cmp-adaptiveform-button__widget"
    >
      <span
        class="cmp-adaptiveform-button__text"
      >
        button
      </span>
    </button>
    <div
      aria-live="polite"
    />
  </div>
</div>
`;
